### PR TITLE
Add support for GitHub teams and team-repos

### PR DESCRIPTION
**Pull request by achantavy**
Testing performed

- Integration tests
- Manually ran CLI

### CODE REVIEW

**Review by resilience-jychp — COMMENTED** [review]
Needed feature, thx for the job.
Some non blocking comments.

May be it can be confusing to have a module with both old way & new model queries ...

File: cartography/models/github/teams.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+from cartography.models.core.relationships import CartographyRelSchema
+from cartography.models.core.relationships import LinkDirection
+from cartography.models.core.relationships import make_target_node_matcher
+from cartography.models.core.relationships import OtherRelationships
+from cartography.models.core.relationships import TargetNodeMatcher
+
+
+@dataclass(frozen=True)
+class GitHubTeamNodeProperties(CartographyNodeProperties):
+    id: PropertyRef = PropertyRef('url')
```
Comment (ramonpetgrave64) [inline cartography/models/github/teams.py:16]: nonblocker: I feel like id should maybe be `{org}/{name}`, but i see that url was also used for the id of Repos

File: tests/integration/cartography/intel/github/test_teams.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+TEST_JOB_PARAMS = {'UPDATE_TAG': TEST_UPDATE_TAG}
+TEST_GITHUB_URL = "https://fake.github.net/graphql/"
+FAKE_API_KEY = 'asdf'
+
+
+@patch.object(cartography.intel.github.teams, '_get_team_repos', return_value=GH_TEAM_REPOS)
+@patch.object(cartography.intel.github.teams, 'get_teams', return_value=GH_TEAM_DATA)
+def test_sync_github_teams(mock_teams, mock_team_repos, neo4j_session):
+    # Arrange
+    _ensure_local_neo4j_has_test_data(neo4j_session)
```
Comment (ramonpetgrave64) [inline tests/integration/cartography/intel/github/test_teams.py:21]: Non blocker: add one more test org to make sure teams aren't accidentally added to that other org.

File: cartography/intel/github/__init__.py [hunk 3]
```diff
@@ -46,5 +47,13 @@ def start_github_ingestion(neo4j_session: neo4j.Session, config: Config) -> None
                 auth_data['url'],
                 auth_data['name'],
             )
+            cartography.intel.github.teams.sync_github_teams(
+                neo4j_session,
+                common_job_parameters,
+                auth_data['token'],
+                auth_data['url'],
+                auth_data['name'],
+                config.update_tag,
```
[the remark this answers is not in the document]
Comment (ramonpetgrave64) [reply cartography/intel/github/__init__.py:56]: I think we can take away this parameter instead, since the update tag is within common_job_parameters.

**Review by ramonpetgrave64 — CHANGES_REQUESTED** [review]

**Review by ramonpetgrave64 — APPROVED** [review]

**Pull request merged**